labrador/src/transcript.rs

```diff
@@ -0,0 +1,77 @@
+// Example of transcript to try the Poseidon sponge
+use crate::poseidon::PoseidonSponge;
+use crate::zq::Zq;
+pub trait Transcript {
+    fn new() -> Self;
+    fn absorb(&mut self, value: Zq);
+    fn get_challenge(&mut self) -> Zq;
+}
+
+// MDS simple example (not cryptographically secure)
+fn simple_mds_like_matrix(size: usize) -> Vec<Vec<Zq>> {
+    let mut matrix = vec![vec![Zq::ZERO; size]; size];
+
+    for (i, row) in matrix.iter_mut().enumerate().take(size) {
+        for (j, elem) in row.iter_mut().enumerate().take(size) {
+            *elem = if i == j {
+                Zq::ONE
+            } else if i + j == size - 1 {
+                Zq::new(2)
+            } else {
+                Zq::new(3)
+            };
+        }
+    }
+    matrix
+}
+
+// ARK simple example (not cryptographically secure)
+fn generate_incremental_round_constants(
+    rounds: usize,
+    state_size: usize,
+    start: u32,
+) -> Vec<Vec<Zq>> {
+    let mut ark = vec![vec![Zq::ZERO; state_size]; rounds];
+    let mut counter = Zq::new(start);
+    for row in ark.iter_mut().take(rounds) {
+        for elem in row.iter_mut().take(state_size) {
+            *elem = counter;
+            counter += Zq::ONE;
+        }
+    }
+
+    ark
+}
+
+pub struct PoseidonTranscript {
+    sponge: PoseidonSponge, // Transcrip will use the Poseidon sponge
+}
+
+impl Transcript for PoseidonTranscript {
+    fn new() -> Self {
+        // Define the Poseidon parameters (just examples for now)
+        let sponge = crate::poseidon::PoseidonSponge::new(
+            8,                                               // rate: usize
+            8,                                               // capacity: usize
+            5,                                               // full_rounds: usize
+            8,                                               // partial_rounds: usize
+            8,                                               // alpha: u64
```

Comment:
The `⁠PoseidonTranscript::new` function hardcodes parameters that should be configurable. Allow parameters to be passed in or use a builder pattern.

```rs
fn new() -> Self {
    Self::with_default_params()
}

fn with_default_params() -> Self {
    Self::with_params(8, 8, 5, 8, 8, 16, 16, 1)
}

fn with_params(
    rate: usize,
    capacity: usize,
    full_rounds: usize,
    partial_rounds: usize,
    alpha: u64,
    matrix_size: usize,
    rounds: usize,
    start: u32,
) -> Self {
    let sponge = crate::poseidon::PoseidonSponge::new(
        rate, capacity, full_rounds, partial_rounds, alpha,
        simple_mds_like_matrix(matrix_size),
        generate_incremental_round_constants(rounds, matrix_size, start),
    );
    Self { sponge }
}
```